### DISCUSSION TITLE
How to set the output size of slides to 1:1(square)?

### DISCUSSION THREAD

**Discussion by samko5sam** [marp-team/marp#151] [Q&A] [answered]
#### Hello 😃 

I'm wondering how to set slides size to `1:1`.

I try to set `size: 1:1` in markdown file, but slides size is still `16:9`. 🤔 

And I try to set `<style>` width:720px , output layout is 720x720px, but image file is 1280x720px, some part of image is blank fill with white

⬇️ markdown file
```markdown

---
marp: true
theme: base
backgroundImage: url('https://marp.app/assets/hero-background.jpg')
paginate: true
size: 16:9
header: header
footer: footer
---

<style>
section{
  width: 720px;
}
</style>

# Slide title
> subtitle


```
⬇️ markdown file marp-cli output image
<img src="https://user-images.githubusercontent.com/75727536/130324390-9e1a2c0a-9ef4-4a5f-9112-8f0232beeaae.png" style="border: 4mm ridge rgba(170, 50, 220, .6);"></img>

Is there any way to set? 😄 Thank you very much

**Comment by yhatt** [comment, marked as answer]
By a technical limitation, overriding `width` and `height` declarations for `section` elements through `<style>` is not working, as you have tried.

You can use custom slide size by using custom theme CSS.

```css
/* @theme square */

@import 'default';

section {
  width: 720px;
  height: 720px;
}
```

```markdown
---
theme: square
---

# Slide title
```

See a documentation of your using tool about how to apply a custom theme CSS: https://github.com/marp-team/marp-cli#use-custom-theme (Marp CLI) and https://github.com/marp-team/marp-vscode#use-custom-theme-css-%EF%B8%8F (Marp for VS Code)

**Reply by imflash217** [reply]
can we apply the `height` and `width` directive per page?
Can we have slides with different page sizes in the same deck?
Thanks

**Reply by imflash217** [reply]
I tried to do this with 
```css
<style scoped>
section {
	height: 1600 px;
	width: 1600 px;
}
</style>
```
Thought the available content size of the that slide changed, but while pdf-export & printing, it still breaks the slide in the original size (thus removing a lot of content during pdf-export & printing.

**Reply by yhatt** [reply]
> It is determined **one size per theme** in Marpit.
> 
> &mdash; https://marpit.marp.app/theme-css?id=slide-size#:~:text=It%20is%20determined%20one%20size%20per%20theme%20in%20Marpit.

Marp cannot change the slide size in the middle of slides by custom class or CSS selector, because the slide size definition is required to be deterministic.

**Reply by yoonjong-park** [reply]
I want to change size, too.
